subsys/fs/nvs/Kconfig

```diff
@@ -0,0 +1,83 @@
+# Kconfig - Non-volatile Storage NVS
+
+#
+# Copyright (c) 2018 Laczen
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+config NVS
+	bool
+	prompt "Non-volatile Storage"
+	default n
+	select FS_FLASH_STORAGE_PARTITION
+	help
+	  Enable support of Non-volatile Storage.
+
+if NVS
+
+
+
+config NVS_PROTECT_FLASH
+	bool
+	prompt "Non-volatile Storage extra flash protection"
+	default n
+	help
+	  Enable extra protection against unnecessary writes to flash. This
+	  enables a extra read check, if data is not changed no write is
+	  performed. If this check is already performed (e.g. no writes unless
+	  data is changed) you can disable this operation.
+
+config NVS_SECTOR_SIZE
```

Comment:
Does it make sense to have `NVS_SECTOR_SIZE`, `NVS_MAX_ELEM_SIZE`, `NVS_SECTOR_COUNT` in `KConfig` when these values are not used by `subsys/fs/nvs` and are specific to a given `struct nvs_fs`? What happen when we have two `struct nvs_fs`?
I would remove them from `KConfig`.

Reply:
@oliviermartin, Thanks for the proposed solutions regarding the SYS_LOG errors, I will change the code according to solution 2.
@oliviermartin, You are right about the settings, I will remove them. I will also remove the use of FLASH_AREA_STORAGE_OFFSET sector and just do a check if:
1. The start it not smaller than FLASH_AREA_STORAGE_OFFSET
2. The end is not larger than FLASH_AREA_STORAGE_OFFSET + FLASH_AREA_STORAGE_SIZE. This would allow multiple struct nvs_fs (but does not check for overlap). The problem I am facing then is how to document it and guide the user to selecting the correct values. Just relying on the sample seems like a bad idea.

Reply:
I am not sure what you mean by `FLASH_AREA_STORAGE_OFFSET`  - I read your document and I understand its definition in a context there is only one partition in one flash device.

But what would be its definition if we have multiple `struct nvs_fs` in multiple flash devices.

On my device I have two flash devices. And I could use NVS on each of these flash devices. So I am not sure having one variable defined as `FLASH_AREA_STORAGE_OFFSET` would work in my context.

Reply:
``FLASH_AREA_STORAGE_OFFSET`` comes from the board definition.
In your case (with 2 flash devices) this will be different and indeed a check would fail. As you know this is not important for NVS. I will remove all the settings and checks for anything in the board definition, if another module would use NVS and would rely on the board definition it just needs to initialize it correct. This way multiple fs in one or multiple flash devices can be handled.